lib/mixins/attributable.js

```diff
@@ -17,10 +17,22 @@ var Attributable = {
         };
       }
 
-      return this.request({
+      var request_opts = {
         method: 'get',
         path: [this.path, id, 'attribute', attribute].join('/')
-      }).then(function (res) {
+      };
+
+      if (alt_auth && alt_auth.token) {
+        request_opts.headers = {
+          'x-auth-token': alt_auth.token
+        };
+      } else if (alt_auth && alt_auth.api_key) {
+        request_opts.headers = {
+          'st2-api-key': alt_auth.api_key
+        };
+      }
```

Comment:
There's a lot of duplication of this code block sprinkle across the code base here.  Is it possible to put this in a reusable function?

Reply:
Fixed.